src/libcommon/utility/utility.cpp

```diff
@@ -652,23 +652,19 @@ size_t CommonUtility::maxPathLength() {
             _maxPathWin = MAX_PATH_LENGTH_WIN_SHORT;
         }
     }
-    return _maxPathWin;
-
+    // For folders in short path mode, it is MAX_PATH - 12
+    // (https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry)
+    // We decided to apply this rule for files too. We could else encounter issues.
+    // If the path of a folder is > MAX_PATH - 12 and the path of a file in this folder is between MAX_PATH - 12 and MAX_PATH.
```

Comment:
```suggestion
    // If the path length of a folder is > MAX_PATH - 12 and the path length of a file in this folder is between MAX_PATH - 12 and MAX_PATH.
```

Reply:
I don't understand how it is possible that `C:\Dir` can be longer than `C:\Dir\file.txt`.

Reply:
**Example Case:**

Let's consider `MAX_PATH` = 24. This means `FOLDER_MAX_PATH` = `MAX_PATH` - 12 = 12.

Now, imagine we have a folder named `C:\test1\test2`, which is 14 characters long. Since this length is greater than `FOLDER_MAX_PATH`, this folder is excluded.

Assuming there's a file named `foo.txt` in this folder, its full path would be `C:\test1\test2\foo.txt`, which is 22 characters long. This length is less than `MAX_PATH`, so the file isn't excluded and will be processed.

The problem arises when we try to synchronize `foo.txt` in a folder that doesn't exist in the subsequent step because we've excluded `C:\test1\test2`

Reply:
Thanks a lot, that's very clear !